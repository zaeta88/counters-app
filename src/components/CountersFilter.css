.btn-filters {
  background: transparent;
  border: none;
  margin: 5px 10px;
}

.btn-filters img {
  height: 25px;
}

.sidebar {
  position: absolute;
}

.sidebar .form-group {
  margin-bottom: 0;
}

.sidebar .form-group .col-md-6 {
  padding-top: 5px;
}

@media (max-width: 415px) {
  .sidebar {
    background-color: #fff;
  }
}