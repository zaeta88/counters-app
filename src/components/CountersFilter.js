import React, { Component } from 'react';
import { Form, Col, Row } from 'react-bootstrap';
import './CountersFilter.css';
import filter from '../assets/img/filter.svg';

class CountersFilter extends Component {
  constructor(props) {
    super(props);
    this.state = {
      visible: false
    };
  }

  handleChange = (event) =>{
    if(event.target.name === 'greater') {
      this.props.countersGreaterThan(event.target.value)
    } else {
      this.props.countersLessThan(event.target.value)
    }   
  };

  toggleMenu = () => {
    this.setState({visible: !this.state.visible})
  }

  render() {
    return (
      <div className="sidebar row">
          <button className="btn btn-filters" onClick={this.toggleMenu}>
            <img src={filter} alt="Filter Icon, Filter counters table."/>
          </button>
          { this.state.visible && 
            <Form.Group as={Row} controlId="validationCount">
              <Col md="6">
                <Form.Control 
                  type="number" 
                  placeholder="Greater Than >" 
                  min='0'
                  name="greater"
                  onChange={this.handleChange}
                />
              </Col>
              <Col md="6">
                <Form.Control 
                  type="number" 
                  placeholder="Less Than <" 
                  min='0'
                  name="less"
                  onChange={this.handleChange}
                />
              </Col>
            </Form.Group>
          }
      </div>
    );
  }
}

export default CountersFilter;