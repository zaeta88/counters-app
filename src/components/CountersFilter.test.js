import React from 'react';
import { shallow } from 'enzyme';
import CountersFilter from './CountersFilter';


describe('React.Component CountersFilter', () => {
  it('should render correctly', () => {
    const component = shallow(<CountersFilter />);
    expect(component).toMatchSnapshot();
  });
});
