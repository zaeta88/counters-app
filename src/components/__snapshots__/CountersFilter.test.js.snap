// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React.Component CountersFilter should render correctly 1`] = `ShallowWrapper {}`;
